engine/env/spark/src/main/java/org/datacleaner/spark/SparkJobContext.java

```diff
@@ -101,20 +103,39 @@ private static Resource createResource(String path) {
     }
 
     public AnalysisJob getAnalysisJob() {
-        if (_analysisJob == null) {
+        return getAnalysisJobBuilder().toAnalysisJob();
+    }
+
+    public AnalysisJobBuilder getAnalysisJobBuilder() {
+        if (_analysisJobBuilder == null) {
             _accumulators.get(ACCUMULATOR_JOB_READS).add(1);
             final Resource analysisJobResource = createResource(_analysisJobPath);
             final DataCleanerConfiguration configuration = getConfiguration();
-            _analysisJob = analysisJobResource.read(new Func<InputStream, AnalysisJob>() {
+            final AnalysisJobBuilder jobBuilder = analysisJobResource.read(new Func<InputStream, AnalysisJobBuilder>() {
                 @Override
-                public AnalysisJob eval(InputStream in) {
+                public AnalysisJobBuilder eval(InputStream in) {
                     final JaxbJobReader jobReader = new JaxbJobReader(configuration);
-                    final AnalysisJob analysisJob = jobReader.read(in);
-                    return analysisJob;
+                    final AnalysisJobBuilder jobBuilder = jobReader.create(in);
+                    return jobBuilder;
                 }
             });
+            _analysisJobBuilder = jobBuilder;
+        }
+        interceptJobBuilder(_analysisJobBuilder, new AtomicInteger(0));
+        return _analysisJobBuilder;
+    }
+
+    private void interceptJobBuilder(AnalysisJobBuilder analysisJobBuilder, AtomicInteger currentComponentIndex) {
```

Comment:
Not the best name here, intercepting (at least to me) is usually when catching calls of some sort, i.e. AOP, or proxying, or putting itself inside a list of callbacks.